SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -807,39 +809,50 @@ void USpatialActorChannel::RemoveRepNotifiesWithUnresolvedObjs(TArray<UProperty*
 	});
 }
 
-void USpatialActorChannel::SpatialViewTick()
+void USpatialActorChannel::ServerCheckIfOwnershipChanged()
 {
-	if (Actor != nullptr && !Actor->IsPendingKill() && IsReadyForReplication())
+	if (!IsAuthoritativeServer())
 	{
-		bool bOldNetOwned = bNetOwned;
+		return;
+	}
+
+	FString NewOwnerWorkerAttribute = improbable::GetOwnerWorkerAttribute(Actor);
 
-		// Use Actor's connection to determine if client owned
-		bNetOwned = false;
-		if (UNetConnection* NetConnection = Actor->GetNetConnection())
+	if (bFirstTick || SavedOwnerWorkerAttribute != NewOwnerWorkerAttribute)
+	{
+		bool bSuccess = Sender->UpdateEntityACLs(GetEntityId(), NewOwnerWorkerAttribute);
+
+		if (bSuccess)
 		{
-			if (APlayerController* PlayerController = NetConnection->PlayerController)
-			{
-				bNetOwned = PlayerController->PlayerState != nullptr;
-			}
+			bFirstTick = false;
+			SavedOwnerWorkerAttribute = NewOwnerWorkerAttribute;
 		}
+	}
+}
 
-		if (bFirstTick || bOldNetOwned != bNetOwned)
-		{
-			if (IsAuthoritativeServer())
-			{
-				bool bSuccess = Sender->UpdateEntityACLs(Actor, GetEntityId());
+void USpatialActorChannel::ClientCheckIfOwnershipChanged()
+{
+	bool bOldNetOwned = bNetOwned;
+	bNetOwned = IsOwnedByWorker();
 
-				if (bFirstTick && bSuccess)
-				{
-					bFirstTick = false;
-				}
-			}
-			else if (!NetDriver->IsServer())
-			{
-				Sender->SendComponentInterest(Actor, GetEntityId());
+	if (bFirstTick || bOldNetOwned != bNetOwned)
+	{
+		Sender->SendComponentInterest(Actor, GetEntityId(), bNetOwned);
+		bFirstTick = false;
+	}
+}
 
-				bFirstTick = false;
-			}
+void USpatialActorChannel::CheckIfOwnershipChanged()
+{
+	if (Actor != nullptr && !Actor->IsPendingKill())
```

Comment:
Could we move the `bFirstTick` check in both the server and client functions to here?

Reply:
To me that seems like it would only complicate things.

Reply:
Ah wait I misread that condition. My bad.